src/server/server.ts

```diff
@@ -9,15 +9,17 @@ type ConnectionSubscriptions = {[reqId: string]: number};
 
 type ConnectionContext = {
   initPromise?: Promise<any>,
-  isLegacy: boolean
+  isLegacy: boolean,
+  socket: WebSocket,
 };
 
-export interface SubscribeMessage {
+export interface RequestMessage {
   [key: string]: any; // any extension that will come with the message.
-  payload: string;
-  query?: string;
-  variables?: {[key: string]: any};
-  operationName?: string;
+  payload: {
```

Comment:
payload can be used to INIT as well.. and is optional (cancel message for example)
also, [key: string] can be removed.